### ISSUE TITLE
Parameter settings in the training phase and How to train the TSN network

### ISSUE THREAD

**Issue by jingoudan** [GuoleiSun/TSNs#1]
Hello author, I recently read your paper and thought about reproducing the network in the paper myself, but your open source code only has the testing phase. I would like to ask you about the training phase strategy in TSNs. There are some questions. For example : 
1、Is task switching performed once per iteration? So is the loss function and optimizer also backward and updated every  batchsize (is trained with task switching)?
2、How is the task embedding network C trained to achieve the third point in the contribution: a small embedding network to learn task conditioning。
I hope to discuss with you more about the content of the article. If it is convenient, we can communicate by email. This is my email: jincancan1998@hnu.edu.cn 
Looking forward to your reply, Thank you ! 谢谢~

**Comment by GuoleiSun** [comment]
Hello, thanks for your interest. Yes, we can discuss more about it. Using email works for me. For your questions, the answers are as follows,
1. Task switching is performed in an epoch-wise manner. That's to say, for each epoch, list of tasks (shuffled) are performed one by one. Optimizer is updated after each iteration.
2. To conduct task i, a task-condition vector for that task is input to the network C, so that the main network is switched to perform task i. The whole process is shown in the Figure 2 of the paper.

You could find more details in the [supplementary](https://openaccess.thecvf.com/content/ICCV2021/supplemental/Sun_Task_Switching_Network_ICCV_2021_supplemental.pdf).
